ocean_lib/data_provider/data_service_provider.py

```diff
@@ -629,8 +675,11 @@ def check_single_file_info(file_url, provider_uri):
         for file_info in response:
             return file_info["valid"]
 
+        return False
+
     @staticmethod
-    def check_asset_file_info(asset, provider_uri):
+    def check_asset_file_info(asset: object, provider_uri: str) -> bool:
```

Comment:
`asset` should be type `Asset`

Reply:
This is a cyclical import issue and we can not apply the same principle as for status_helper, since this is clearly a separate class with separate concerns. What do you recomend? `object` or `Any`?

Reply:
I recommend we modify the logic to take a DID `str` like so:

```python
    @staticmethod
    def check_asset_file_info(did: str, provider_uri: str) -> bool:
        """Check if an asset's file info is valid. Specify asset by its DID."""
        if not did:
            return False
        _, endpoint = DataServiceProvider.build_fileinfo(provider_uri)
        data = {"did": did}
        response = requests.post(endpoint, json=data)

        if response.status_code != 200:
            return False

        response = response.json()
        for ddo_info in response:
            return ddo_info["valid"]

        return False
```